### DISCUSSION TITLE
Load search engine code only if necessary

### DISCUSSION THREAD

**Discussion by CharlesNepote** [quarto-dev/quarto-cli#9604] [Feature Requests]
### Description

I would like to [improve first page load](https://github.com/quarto-dev/quarto-cli/discussions/9330), and I guess the majority of the users are rarely using the search engine, if ever. This one is currently always loaded: quarto-search.js and autocomplete.js represent 37KB + 93KB = 130 KB which are useless most of the time.

By the way, it seems possible to launch the search engine on demand (using events such as onmouseover, onfocus...). This article is showing the way: https://hacks.mozilla.org/2021/08/mdns-autocomplete-search/

**Comment by cscheid** [comment]
These are shared and cached across all pages in a website, so although I sympathize with the overall goal, the gain is not as large as you'd expect. We have other things higher in our priority list, so don't expect that fix in the near or medium term.

**Reply by CharlesNepote** [reply]
Yes I perfectly understand it is not a priority.

I also understand that the gain is not very important. But it's also a question of changing our mindset. The internet were not so bad when pages weighted less than 100KB ;-)

**Reply by cscheid** [reply]
> I also understand that the gain is not very important. But it's also a question of changing our mindset.

I know you're trying to be persuasive, but it comes across as condescending. Consider instead that we share the mindset and making those choices in awareness. We only have so many hours in the day, and the other fixes make the internet better. You are always welcome to submit a PR instead of a discussion.
